server/models/shape/__init__.py

```diff
@@ -118,6 +119,52 @@ def center_at(self, x: int, y: int) -> None:
     def subtype(self):
         return getattr(self, f"{self.type_}_set").get()
 
+    def make_copy(self, dst_layer):
+        new_shape = Shape.create(
+            uuid=str(uuid4()),
+            layer=dst_layer,
+            type_=self.type_,
+            x=self.x,
+            y=self.y,
+            name=self.name,
+            name_visible=self.name_visible,
+            fill_colour=self.fill_colour,
+            stroke_colour=self.stroke_colour,
+            vision_obstruction=self.vision_obstruction,
+            movement_obstruction=self.movement_obstruction,
+            is_token=self.is_token,
+            annotation=self.annotation,
+            draw_operator=self.draw_operator,
+            index=self.index,
+            options=self.options,
+            badge=self.badge,
+            show_badge=self.show_badge,
+            default_edit_access=self.default_edit_access,
+            default_vision_access=self.default_vision_access,
+            is_invisible=self.is_invisible,
+            default_movement_access=self.default_movement_access,
+            is_locked=self.is_locked,
+            angle=self.angle,
+            stroke_width=self.stroke_width,
+            asset=self.asset,
+            group=None if self.group is None else self.group.make_copy(),
```

Comment:
This creates a new group for each member. So for example a group with 3 shapes will result in 3 separate groups

Reply:
This hasn't been addressed yet

Reply:
Yeah, had a couple busy weekends, and it took a bit to come up with a good way to make this work. Just pushed something up that takes care of it.